Restore docs: updated restore tables documentation (#3490)
scylladb/scylla-manager: docs/source/restore/restore-tables.rst

.. note:: Currently, Scylla Manager does not support restoring content of `CDC log tables <https://docs.scylladb.com/stable/using-scylla/cdc/cdc-log-table.html>`_.

.. note::

>    The ``--keyspace`` flag defines base tables which will be restored.

Reviewer: ```suggestion
   The ``--keyspace`` flag defines base tables that will be restored.
```

Reviewer: The content of this note is clear, but the usage is not so much. 
Is the user expected to run:
`sctool restore --cluster <id|name> --location [<dc>:]<provider>:<bucket> --snapshot-tag <tag> [flags] ---restore-tables --keyspace`
?
Is adding `--keyspace` required?

I THINK this section should be modified like this:

- (If --keyspace is optional)

_To restore the content of the tables (rows), use the :ref:`sctool restore <sctool-restore>` command with the ``--restore-tables`` flag._

_You **can** add the ``--keyspace`` flag to define the base tables that will be restored.  All Materialized Views and Secondary Indexes of those tables will be automatically recreated in order to ensure the correct data state._

- (If --keyspace is mandatory)

_To restore the content of the tables (rows), use the :ref:`sctool restore <sctool-restore>` command with the ``--restore-tables`` flag._

_**Add** the ``--keyspace`` flag to define the base tables that will be restored.  All Materialized Views and Secondary Indexes of those tables will be automatically recreated in order to ensure the correct data state._

   All Materialized Views and Secondary Indexes of those tables will be automatically recreated in order to ensure correct data state.

To restore the content of the tables (rows), use the :ref:`sctool restore <sctool-restore>` command with the ``--restore-tables`` flag.

Prerequisites
=============

> * Scylla Manager requires CQL credentials with:

Reviewer: ```suggestion
* Scylla Manager requires CQL credentials with:

```

Reviewer: You must add an empty line for this section to render correctly.

    * `permission to alter <https://opensource.docs.scylladb.com/stable/operating-scylla/security/authorization.html#permissions>`_ restored tables.
    * `permission to drop and create <https://opensource.docs.scylladb.com/stable/operating-scylla/security/authorization.html#permissions>`_ Materialized Views and Secondary Indexes of restored tables.

> * Restoring the content of the tables assumes that the correct schema (identical as in the backup) is already present in the destination cluster.

Reviewer: ```suggestion
* Restoring the content of the tables assumes that the correct schema (identical as in the backup) is already present in the destination cluster.

```

Reviewer: You must add an empty line for this section to render correctly.

   Scylla Manager does NOT validate that, so it's user responsibility to ensure it. The only form of validation
   that Scylla Manager performs is checking whether restored tables are present in destination cluster,
   but it does not validate their columns types nor other properties. In case destination cluster is missing correct schema,
   it should be :doc:`restored <restore-schema>` first.

>    It's important to note, that this requirement also involves keyspace schema.

Reviewer: ```suggestion
   Note that this requirement also applies to the keyspace schema.
```

>    Keyspace (where restored tables live) should have correct schema, even in terms of replication strategy.

Reviewer: ```suggestion
   Keyspace (where restored tables live) should have the correct schema, including the replication strategy.
```
